.github/workflows/spec-update.yaml

```diff
@@ -0,0 +1,168 @@
+
+jobs:
+  generate:
+    name: "Download, Generate, Compile and Push"
+    runs-on: [ubuntu-latest]
+    permissions:
+      pull-requests: write
+      contents: write
+    outputs:
+      spec_diff: ${{ steps.spec_diff.outputs.spec_diff }}
+      branch: ${{ steps.push.outputs.branch }}
+      compilation_result: ${{ steps.compile.outputs.compilation_result }}
+      test_result: ${{ steps.compile.outputs.test_result }}
+    env:
+      API_BASE_URL: "https://github.tools.sap/api/v3/repos"
+      CHOICE: ${{ github.event.inputs.file }}
+      REF: ${{ github.event.inputs.file-ref }}
+      CREATE_PR: ${{ github.event.inputs.create-pr }}
+    steps:
+      - name: "Checkout repository"
+        uses: actions/checkout@v4
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: "Checkout or Create Branch"
+        id: branch
+        # Checkout branch if it exists, otherwise create it
+        run: |
+          BRANCH="spec-update/$CHOICE/$REF"
+          git fetch origin $BRANCH || true
+          git checkout -B $BRANCH origin/$BRANCH || git checkout -b $BRANCH
+          echo "branch=$BRANCH" >> "$GITHUB_OUTPUT"
+
+      - name: "Download specification file"
+        id: download
+        env:
+          GH_ENTERPRISE_TOKEN: ${{ secrets.GH_TOOLS_TOKEN }}
+        run: |
+          case $CHOICE in
+            core)
+              API_URL="$API_BASE_URL/cloudsdk/cloud-sdk-java-tests/contents/aicore.yaml?ref=$REF"
+              FILE_PATH='packages/ai-api/src/spec/AI_CORE_API.yaml'
+              ;;
+            grounding)
+              # TODO
+              exit 1
+              ;;
+            orchestration)
+              API_URL="$API_BASE_URL/AI/llm-orchestration/contents/src/spec/api.yaml?ref=$REF"
+              FILE_PATH='packages/orchestration/src/spec/api.yaml'
+              ;;
+          esac
+          
+          echo "Downloading $CHOICE specification file from $API_URL ..."
+          
+          gh api $API_URL -H "Accept: application/vnd.github.raw" > $FILE_PATH
+
+      - name: "Exit if there are no changes"
+        id: spec_diff
+        run: |
+          if [[ `git status --porcelain` ]]; then
+            echo "spec_diff=true" >> "$GITHUB_OUTPUT"
+          else
+            echo "spec_diff=false" >> "$GITHUB_OUTPUT"
+          fi
+
+      - uses: sap/ai-sdk-js/.github/actions/setup@main
+        if: steps.spec_diff.outputs.spec_diff == 'true'
+        with:
+          node-version: 20
+
+      - name: "Generate"
+        id: generate
+        if: steps.spec_diff.outputs.spec_diff == 'true'
+        run: |
+          pnpm generate
+
+      - name: "Compile and Test"
+        id: compile
+        if: steps.spec_diff.outputs.spec_diff == 'true'
+        # Compilation can easily fail e.g. from re-namings and has to be fixed manually.
+        # Thus, this action raises the PR anyway and only reports success or failure of compilation and testing.
+        run: |
+          if pnpm compile ; then
```

Comment:
[req] Here need to run `pnpm install` first to install (dev-)dependencies, which will call `pnpm compile` afterwards anyway.